core/arch/arm/include/sm/optee_smc.h

```diff
@@ -559,6 +559,37 @@
 /* See OPTEE_SMC_CALL_WITH_REGD_ARG above */
 #define OPTEE_SMC_FUNCID_CALL_WITH_REGD_ARG	U(19)
 
+/*
+ * Access to the secure watchdog.
+ *
+ * Call requests usage:
+ * a0	SMC Function ID, OPTEE_SMC_FUNCID_WATCHDOG
+ * a1	Watchdog command
+ *	0 : Init watchdog
+ *	1 : Set timeout
+ *	2 : Enable watchdog
+ *	3 : Ping the watchdog for refresh
+ *	4 : Get time left
+ * a2-6	Not use
+ * a7	Hypervisor Client ID register
+ *
+ * Normal return register usage:
+ * a0	OPTEE_SMC_RETURN_OK
```

Comment:
a0 may also be `PSCI_RET_INTERNAL_FAILURE`, `PSCI_RET_INVALID_PARAMETER` or `PSCI_RET_NOT_SUPPORTED`.

Reply:
Should this description states that RET_OK and UNKNOWN_FONCTION IDs are SMCCC standard values that make this OPTEE_SMC_RETURN/PSCI_RET mix to be consistent?

```
 * Normal return register usage:
 * a0	PSCI_RET_SUCCESS
 * a1-7	Preserved
 *
 * Error return:
 * a0	PSCI_RET_NOT_SUPPORTED      Function not supported
 * 	PSCI_RET_INVALID_PARAMETERS Invalid arguments
 * 	PSCI_RET_INTERNAL_FAILURE   Device error
 * a1-7	Preserved
 *
 * By SMCCC convention:
 * PSCI_RET_SUCCESS, OPTEE_SMC_RETURN_OK and ARM_SMCCC_RET_SUCCESS
 * are equal.
 * PSCI_RET_NOT_SUPPORTED, OPTEE_SMC_RETURN_UNKNOWN_FUNCTION and  
 * ARM_SMCCC_RET_NOT_SUPPORTED are equal.
 */

Reply:
Looks good to me to mix both PSCI and OPTEE return args., let's follow that.